integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -0,0 +1,316 @@
+        for (int index = 0; index < in.limit(); index++) {
+            byte b = in.get(index);
+            byte rotated = (byte) (b + rot);
+            // System.out.println("rotate(" + b + ", " + rot + "): " + rotated);
+            out.put(index, rotated);
+        }
+        return out;
+    }
+
+    @Test
+    public void shouldPassThroughRecordUnchanged() throws Exception {
+        String proxyAddress = "localhost:9192";
+
+        try (var cluster = ClusterFactory.create(ClusterConfig.builder().build())) {
+            cluster.start();
+
+            String bootstrapServers = cluster.getBootstrapServers();
+            try (var admin = Admin.create(Map.of(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers))) {
+                admin.createTopics(List.of(new NewTopic(TOPIC_1, 1, (short) 1))).all().get();
+            }
+
+            String config = """
+                    proxy:
+                      address: %s
+                    clusters:
+                      demo:
+                        bootstrap_servers: %s
+                    filters:
+                    - type: ApiVersions
+                    - type: BrokerAddress
+                    """.formatted(proxyAddress, cluster.getBootstrapServers());
+
+            var proxy = startProxy(config);
+
+            var producer = new KafkaProducer<String, String>(Map.of(
+                    ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress,
+                    ProducerConfig.CLIENT_ID_CONFIG, "shouldPassThroughRecordUnchanged",
+                    ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class,
+                    ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG, 3_600_000));
+            producer.send(new ProducerRecord<>(TOPIC_1, "my-key", "Hello, world!")).get();
+            producer.close();
+
+            var consumer = new KafkaConsumer<String, String>(Map.of(
+                    ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, proxyAddress,
+                    ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                    ConsumerConfig.GROUP_ID_CONFIG, "my-group-id",
+                    ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest"));
+            consumer.subscribe(Set.of(TOPIC_1));
+            var records = consumer.poll(Duration.ofSeconds(10));
+            consumer.close();
+            assertEquals(1, records.count());
+            assertEquals("Hello, world!", records.iterator().next().value());
+
+            // shutdown the proxy
+            proxy.shutdown();
+
+        }
+
+    }
+
+    @Test
+    public void shouldModifyProduceMessage() throws Exception {
+        String proxyAddress = "localhost:9192";
+        try (var cluster = ClusterFactory.create(ClusterConfig.builder().build())) {
```

Comment:
Any reason to create the cluster in each test method rather than in `beforeEach` (cooler yet would be using a junit extension 😉)?